### PR TITLE
Remove redundant `Connection#connect()` calls

### PR DESCRIPTION
**Pull request by rosier**
This PR removes a few unneeded calls to `Connection::connect()`

### CODE REVIEW

File: lib/Doctrine/DBAL/Connection.php [hunk 1]
```diff
@@ -771,14 +769,12 @@ public function fetchAll($sql, array $params = array(), $types = array())
...
      * @param string $statement The SQL statement to prepare.
      *
-     * @return \Doctrine\DBAL\Driver\Statement The prepared statement.
+     * @return \Doctrine\DBAL\Statement The prepared statement.
      *
      * @throws \Doctrine\DBAL\DBALException
      */
     public function prepare($statement)
     {
-        $this->connect();
```
Comment (stof) [inline lib/Doctrine/DBAL/Connection.php]: are you sure about that ? I'm quite sure that preparing a statement requires being connected

Comment (rosier) [inline lib/Doctrine/DBAL/Connection.php]: Yes, `Statement::__construct` also calls `Connection::connect()` via `Connection::getWrappedConnection()`

**Comment by Ocramius** [comment]
Looks sane to me /cc @deeky666

**Comment by stof** [comment]
:+1:

**Comment by deeky666** [comment]
@rosier thanks!

**Pull request merged**